File: typegraph/python_next/typegraph/wit.py
# Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
# SPDX-License-Identifier: MPL-2.0

from wasmtime import Store
from typegraph.gen import TypegraphCore
from typegraph.gen.exports.core import Core
from typegraph.gen.exports.runtimes import Runtimes


store = Store()
_typegraph_core = TypegraphCore(store)

core = Core(_typegraph_core)
runtimes = Runtimes(_typegraph_core)

File: typegraph/node/sdk/src/wit.ts
// Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
// SPDX-License-Identifier: MPL-2.0

import { MetatypeTypegraphCore } from "./gen/interfaces/metatype-typegraph-core.js";
import { MetatypeTypegraphRuntimes } from "./gen/interfaces/metatype-typegraph-runtimes.js";
import { MetatypeTypegraphAws } from "./gen/interfaces/metatype-typegraph-aws.js";
import { MetatypeTypegraphUtils } from "./gen/interfaces/metatype-typegraph-utils.js";
import * as js from "./gen/typegraph_core.js";

export const core = js.core as typeof MetatypeTypegraphCore;
export const runtimes = js.runtimes as typeof MetatypeTypegraphRuntimes;
export const aws = js.aws as typeof MetatypeTypegraphAws;
export const wit_utils = js.utils as typeof MetatypeTypegraphUtils;

export type { Cors, Rate } from "./gen/interfaces/metatype-typegraph-core.js";
export type {
  Auth,
  AuthProtocol,
  AuthProtocolBasic,
  AuthProtocolJwt,
  AuthProtocolOauth2,
} from "./gen/interfaces/metatype-typegraph-utils.js";
